api/v1beta1/nutanixcluster_types.go

```diff
@@ -145,7 +147,22 @@ func (ncl *NutanixCluster) SetConditions(conditions capiv1.Conditions) {
 	ncl.Status.Conditions = conditions
 }
 
-//+kubebuilder:object:root=true
+func (ncl *NutanixCluster) GetCredentialRefForCluster() (*credentialTypes.NutanixCredentialReference, error) {
```

Comment:
I think it makes sense for this to be a method of the struct.

Reply:
Consider streamlining the name while at it: Something like `PrismCredentialRef` or `GetPrismCredentialRef` stutters less.

Reply:
🤦‍♂️ forgot to actually do what I left a comment on, fixed in https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/360/commits/d8b6f454525affd1c7cd5347182fc2614d4207a5